clinica/utils/filemanip.py

```diff
@@ -463,9 +455,53 @@ def extract_metadata_from_json(json_file: str, list_keys: List[str]) -> List[str
             list_values.append(data[key])
         else:
             missing_keys.append(key)
-    if len(missing_keys) > 0:
-        error_message = f"Clinica could not find the following keys in the following JSON file: {json_file}:\n- "
-        error_message += "\n- ".join(missing_keys)
-        raise ClinicaException(error_message)
+    if len(missing_keys) > 0 and handle_missing_keys is None:
+        raise ClinicaException(
+            f"Clinica could not find the following keys in the following JSON file: {missing_keys}."
+        )
+    if handle_missing_keys:
+        patch = handle_missing_keys(data, missing_keys)
+        data.update(patch)
+    return [data[k] for k in list_keys]
+
 
-    return list_values
+def _handle_missing_keys_dwi(data: dict, missing_keys: set) -> dict:
+    handlers = {
+        "TotalReadoutTime": _handle_missing_total_readout_time,
+        "PhaseEncodingDirection": _handle_missing_phase_encoding_direction,
+    }
+    try:
+        return {k: handlers[k](data, missing_keys) for k in missing_keys}
+    except KeyError:
+        raise ValueError(
+            f"Could not recover the missing keys {missing_keys} from JSON file."
+        )
+
+
+def _handle_missing_total_readout_time(data: dict, missing_keys: set) -> float:
```

Comment:
Could you add a docstring describing the strategy you implemented to recover missing readout time ?